components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -227,7 +227,18 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
                 if (request.getAttribute(FrameworkConstants.RESTART_LOGIN_FLOW) != null &&
                         request.getAttribute(FrameworkConstants.RESTART_LOGIN_FLOW).equals("true")) {
                     context = (AuthenticationContext) context.getProperty(FrameworkConstants.INITIAL_CONTEXT);
+                    if (context == null) {
```

Comment:
This is happening coz we dont have the initial context clone inside the context after switch the browser for the 1st time. Instead of sending an error message like this we may be able to handle this internally by cloning the initial context and setting it when we retrieve the context after a browser switch

Reply:
This issue will only come if there is a second browser switch right?
If that's the case lets remove this improvement from this PR and check on handling it in the way i mentioned in the previous comment

Reply:
`This issue will only come if there is a second browser switch right?`
Yes.
`If that's the case lets remove this improvement from this PR and check on handling it in the way i mentioned in the previous comment`
Sure. I'll remove this from this PR and send another one by properly handing this

Reply:
Fixed with ae9e3f2a4fbe6171b1e508f873d6c19ceba817cd